src/logic/calculate.js

```diff
@@ -0,0 +1,57 @@
+import operate from './operate';
+
+const calculate = (obj, buttonName) => {
+  const objData = obj;
```

Comment:
You can destructure `obj` at once
`const {total, next, operation }  = obj;    `